src/consumer/mod.rs

```diff
@@ -199,6 +247,19 @@ impl<E> ConsumerBuilder<E> {
         self
     }
 
+    /// Register a handler for the given job type (`kind`).
+    ///
+    /// Whenever a job whose type matches `kind` is fetched from the Faktory, the given handler
+    /// object is called with that job as its argument.
+    pub fn register_runner<K, H>(&mut self, kind: K, runner: H) -> &mut Self
```

Comment:
I think you can change `register` to just call this method internally. We only really need it to give the type checker a boost to realize the required closure type.